src/DataSets/log.js

```diff
@@ -1,44 +1,73 @@
 import React from 'react';
 import { getInstance as getD2 } from 'd2/lib/d2';
 
+const maxLogsPerPage = 200;  // TODO: maybe make it readable from the dataStore
+const maxLogPages = 100;
 
-async function log(actionName, status, dataset) {
+
+async function log(actionName, status, datasets) {
     // Log the name of the action that has been executed, its status
     // ("success", "failed"), by whom and on which datasets.
-    const maxLogs = 1e3;
-
-    const d2 = await getD2();
-    const store = await d2.dataStore.get('dataset-configuration');
-
-    const logIndex = await store.get('logIndex').catch(() => 0);
-    const logs = await store.get('logs').catch(() => []);
-
-    const datasets = Array.isArray(dataset) ? dataset : [dataset];
+    const store = await getStore();
+    const logs = await getLogs([0], store);
+    logs.push(await logEntry(actionName, status, datasets));
+    setLogs(logs, store);
+}
 
-    const newLog = {
+async function logEntry(actionName, status, datasets) {
```

Comment:
There are two functions called logEntry in this file

Reply:
Yes, one `logEntry` (creates a new log entry object) and a function-component `LogEntry` (capital `L`), that displays a log entry.

I'm changing the first one to `makeEntry` to avoid confusion.

Reply:
Perfect!